### PR TITLE
Fix for Add WEBHOOK_JOB to GHEvent

### LINKED ISSUE #1288: GHEvents enum is missing workflow_job
**Describe the bug**
I’m trying to set up a webhook for a GitHub application which responds to the following events: create, delete, github_app_authorization, ping, workflow_dispatch, workflow_job, and workflow_run.  However, the code failed when it tried to find the GHEvent enum value for “workflow_job”.  I don’t know exactly what the difference between a “job” and a “run” is, but I can’t very well find out without setting up a webhook for both.

**To Reproduce**
```
            String event = "webhook_job";
            boolean found = false;
            for (GHEvent ghEvent : GHEvent.values()) {
                if (ghEvent.toString().equalsIgnoreCase(event)) {
                    found = true;
                    break;
                }
            }
            if (!found)
                throw new IllegalArgumentException("Invalid webhook event name: " + event);
```

**Expected behavior**
`found == true`

**Desktop (please complete the following information):**
 - Java: 1.8
 - Version 1.133

**Additional context**
https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_job

### PR DESCRIPTION
**Pull request by katieshin**
# Description

Adding fix for https://github.com/hub4j/github-api/issues/1288 - new enum value for WORKFLOW_JOB in GHEvent.

Fixes #1288 

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time.

- [ ] Push your changes to a branch other than `main`. Create your PR from that branch.
- [ ] Add JavaDocs and other comments
- [ ] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [ ] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [ ] Fill in the "Description" above.
- [ ] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHEvent.java [hunk 1]
```diff
@@ -69,6 +69,7 @@
     TEAM,
     TEAM_ADD,
     WATCH,
+    WEBHOOK_JOB,
```
Comment (gsmet) [inline src/main/java/org/kohsuke/github/GHEvent.java:72]: It's not `WEBHOOK_JOB` but `WORKFLOW_JOB`.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: src/main/java/org/kohsuke/github/GHEvent.java (hunk 1 above)
Comment (katieshin) [reply src/main/java/org/kohsuke/github/GHEvent.java:72]: Oh! I'll make the change, thanks.

Comment (gsmet) [reply src/main/java/org/kohsuke/github/GHEvent.java]: Also, my guess is that you will need to change things here: https://github.com/hub4j/github-api/blob/main/src/test/java/org/kohsuke/github/EnumTest.java#L31 .

Always run the tests after a change, especially in a code base you don't know as it's always full of surprises :).

Comment (katieshin) [reply src/main/java/org/kohsuke/github/GHEvent.java]: Thank you for the help!

**Review by gsmet — APPROVED** [review]
Maybe this should be squashed into one commit for a cleaner history before merging?

**Pull request merged**